docs/applications_general.py

```diff
@@ -0,0 +1,190 @@
+#In this example, we use a Python data flow scripting interface to connect
+#PyEmma, a simulator adapter, and some additional Python code to iteratively
+#run and analyze groups of simulations until convergence.
+#
+#The main script body defines and executes the subgraph.
+#
+#One supporting module (`analyze.py`) wraps PyEmma.
+#
+#supporting modules (`gromacs_wrapper.py` and `lammps_wrapper.py`) 
+#wraps command line MD commands with a SCALE-MS wrapper layer.  The results should be in the same format.
+
+from gromacs_wrapper import collect_configurations, structure_to_pdb, make_input, simulate, modify_input
+
+# configuration_list (an array of filenames) and md_inputs (a file
+# name, or list of file names if multiple inputs are needed to
+# intitialize the simulation.
+
+
+coordinate_inputs = configuration_input_list # some list of files
+run_parameters = md_inputs # some list of files
+topology_files = topology_input_list # some list of files
+
+initial_simulation_input = make_input(
+    simulation_parameters=run_parameters,
+    topology=topology_file,
+    conformation=coordinate_inputs)
+
+# Set up an array of N simulations, starting from a single input.
+# MRS: seems like some instructions would need to be given here, such as:
+#      - which entries in the files to replaced by random numbers, or state dependent inputs
+#      - User should not have to worry about details such as whether temporary files would be needed should be 
+#      - so maybe the broadcasating would be done in make_input
+#      - 
+initial_input = scalems.broadcast(initial_simulation_input, shape=(N,))
+
+# We will need a pdb for MSM building in PyEmma
+initial_pdb = coordinate_inputs[0]
+
+# Get a placeholder object that can serve as a sub context / work graph owner
+# and can be used in a control operation.
+
+# MRS: I'm not sold on using the name subgraph, because it doesn't
+#      describe what it is a subgraph of. Can this be made more specific?
+
+simulation_and_analysis_iteration = scalems.subgraph(variables={
+        'conformation': initial_input,
+        'transition_matrix': scalems.ndarray(0., shape=(N, N)),  # number of simulations or number of clusters?
+        'is_converged': False})
+
+with simulation_and_analysis_iteration:
+    modified_input = modify_input(
+        input=initial_input, structure=simulation_and_analysis_iteration.conformation)
+    md = simulate(input=modified_input)
+    
+    # Get the output trajectories and pass to PyEmma to build the MSM
+    # Return a stop condition object that can be used in gmx while loop to
+    # terminate the simulation
+    allframes = collect_configurations(md.output.trajectory)
+    
+    adaptive_msm = analysis.msm_analyzer(molecular_topology=initial_pdb,
```

Comment:
To do: consider how to express this more generically in terms of accumulating some number of samples from some number of simulations to cluster some number of model states to propose some number of follow-up simulations, where these numbers are decoupled and where the model accumulates data. Consider the constraints, if any, on whether batches of samples can be provided in arbitrary order or in arbitrary sized batches. This then leads to a generalization such that the algorithm can be expressed in various asynchronous ways, and we can explore iterator patterns.